climada/hazard/tc_tracks.py

```diff
@@ -2980,3 +2980,176 @@ def _zlib_from_dataarray(data_var: xr.DataArray) -> bool:
+
+    Parameters:
+    ----------
+    tc_track: TCT track object
+        track object containing a list of all tracks
+    res: int (optional), default: 5°
+        resolution in degrees of the grid bins in which the density will be computed
+    genesis: bool, Default = False
+        If true the function computes the track density of only the genesis location of tracks
+    norm: bool (optional), default = False
+        If False it returns the number of samples in each bin. If True, returns the
+        specified normalization function at each bin computed as count_bin / grid_area.
+    filter_tracks: bool (optional) default: True
+        If True the track density is computed as the number of different tracks crossing a grid
+        cell. If False, the track density takes into account how long the track stayed in each
+        grid cell. Hence slower tracks increase the density if the parameter is set to False.
+    wind_min: float (optional), default: None
+        Minimum wind speed above which to select tracks.
+    wind_max: float (optional), default: None
+        Maximal wind speed below which to select tracks.
+    Returns:
+    -------
+    hist_count: np.ndarray
+        2D matrix containing the the absolute count per gridd cell of track point or the normalized
+        number of track points, depending on the norm parameter.
+
+    Example:
+    --------
+    >>> tc_tracks = TCTrack.from_ibtracs_netcdf("path_to_file")
+    >>> tc_tracks.equal_timestep(time_steph_h = 1)
+    >>> hist_count, *_ = compute_track_density(res = 1)
+    >>> plot_track_density(hist_count)
+
+    """
+
+    limit_ratio: float = 1.12 * 1.1  # record tc speed 112km/h -> 1.12°/h + 10% margin
+    time_value: float = tc_track.data[0].time_step[0].values.astype(float)
+
+    if time_value > (res / limit_ratio):
+        warnings.warn(
+            "The time step is too big for the current resolution. For the desired resolution, \n"
+            f"apply a time step of {res/limit_ratio}h."
+        )
+    elif res < 0.1:
+        warnings.warn(
+            "The resolution is too high. The computation might take several minutes \n"
+            "to hours. Consider using a resolution below 0.1°."
+        )
+
+    # define grid resolution and bounds for density computation
+    lat_bins: np.ndarray = np.linspace(-90, 90, int(180 / res))
+    lon_bins: np.ndarray = np.linspace(-180, 180, int(360 / res))
+    # compute 2D density
+    if genesis:
+        hist_count = compute_genesis_density(
+            tc_track=tc_track, lat_bins=lat_bins, lon_bins=lon_bins
+        )
+    else:
+        hist_count = np.zeros((len(lat_bins) - 1, len(lon_bins) - 1))
+        for track in tqdm(tc_track.data, desc="Processing Tracks"):
+
+            # select according to wind speed
+            wind_speed = track.max_sustained_wind.values
+            if wind_min and wind_max:
+                index = np.where((wind_speed >= wind_min) & (wind_speed <= wind_max))[0]
```

Comment:
Personally I'd use `>= wind_min` but `< wind_max` (here and below), such that if you call the function once with `wind_min=10, wind_max=20` and then with `wind_min=20` you don't get track points with a wind speed of exactly 20 counted in both. In other words I'd make `wind_min` inclusive but `wind_max` exclusive (i.e., `[wind_min, wind_max)`).

Reply:
Good point!